include/triton/core/tritonbackend.h

```diff
@@ -1411,6 +1411,36 @@ TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_GetBackendAttribute(
     TRITONBACKEND_Backend* backend,
     TRITONBACKEND_BackendAttribute* backend_attributes);
 
+/// Query the total memory usage of the model and its associated model
+/// instances. Unless TRITONBACKEND_BackendAttributeSetEnableMemoryTracker is
+/// explicitly set to true, this function is optional, a backend is not required
+/// to implement it. This function is called when changes are made to the model
+/// and Triton wants to retrieve the latest memory usage. This function may be
+/// called multiple times within the lifetime of the model (between
+/// TRITONBACKEND_ModelInitialize and TRITONBACKEND_ModelFinalize).
+/// The backend may return error to indicate failure to get the latest memory
+/// usage.
+///
+/// \param model The model.
+/// \param usage The list of buffer attributes that the backend can record the
+/// memory usage, each entry should record the total memory usage of a given
+/// memory type and id. For example, if the model itself occupies 64 bytes on
+/// CUDA device 0 and it has two model instances that each instance occupies
```

Comment:
Question: Should the api be in terms of model instances or models? Do we need to specify this is for GPU memory - or does it include cpu memory as well?